src/components/Results.js

```diff
@@ -1,46 +1,73 @@
 
-import React from 'react';
+import React, { Component } from 'react';
 import moment from 'moment';
 
-const Results = (props) => {
-  return (
-    <table className="table">
-    <thead>
-      <tr>
-
-        <th>title</th>
-        <th>firstname</th>
-        <th>surname</th>
-        <th>email</th>
-        <th>roomId</th>
-        <th>checkInDate</th>
-        <th>checkOutDate</th>
-        <td>Durations stayed in</td>
-      </tr>
-      </thead>
-      <tbody>
-      {props.results.map(result => {
-      
-        return (
+class Results extends Component {
+  
+  state={
+    selected:[],
+    row:[],
+    results:this.props.results,
+    isRowToggled:false
+  }
+
+  handleClick =(e)=>{
+    // if(handleClick%2==0){
+    //   return this.state.onClick
+    // } else{
+    //   return this.state.default
+    // }
+    this.setState({
+      isRowToggled:!this.state.isRowToggled
+     
+    })
+
+  }
+  render() {
+    
+   
+    return (
+      <table className="table">
+        <thead>
           <tr>
 
-
-            <td>{result.title}</td>
-            <td>{result.firstName}</td>
-            <td>{result.surname}</td>
-            <td>{result.email}</td>
-            <td>{result.roomId}</td>
-            <td>{result.checkInDate}</td>
-            <td>{result.checkOutDate}</td>
-            <td>{moment([result.checkOutDate]).diff(moment([result.checkInDate]), 'days')}</td>
-            
+            <th>title</th>
+            <th>firstname</th>
+            <th>surname</th>
+            <th>email</th>
+            <th>roomId</th>
+            <th>checkInDate</th>
+            <th>checkOutDate</th>
+            <td>Durations stayed in</td>
           </tr>
-        )
-      })}
-      </tbody>
-    </table>
-
-  )
+        </thead>
+        <tbody>
+          {this.state.results.map(result => {
+
+            return (
+              <tr 
+                onClick={this.handleClick}
+                className={this.state.isRowToggled ? "onClick" : null}
```

Comment:
I think we discussed in class last week the solution to the problem where the toggle state is applied to all rows not just the one you click on. To recap: you need to extract each row as it's own component. Then each of these row components can hold it's own individual state.